src/tools/build-manifest/src/manifest.rs

```diff
@@ -111,6 +139,27 @@ impl Serialize for FileHash {
     }
 }
 
+fn tarball_variant(builder: &mut Builder, base: &Path, ext: &str) -> Option<PathBuf> {
+    let mut path = base.to_path_buf();
+    path.set_extension(ext);
```

Comment:
Where does the .tar come in here?

Reply:
The function receives `foo.tar.gz`, and the stdlib's `Path` only treats `gz` as its extension. So, when the function is called with `ext = "gz"` this line does nothing, and when it's called with `ext = "xz"` it replaces `foo.tar.gz` with `foo.tar.xz`.

Reply:
Ok. I guess probably not worth a comment.